hail/src/main/scala/is/hail/expr/ir/IRBuilder.scala

```diff
@@ -310,18 +315,18 @@ object IRBuilder {
     def apply(body: IRProxy): IRProxy = in(body)
 
     def in(body: IRProxy): IRProxy = { (env: E) =>
-      LetProxy.bind(bindings, body, env, isAgg = true)
+      LetProxy.bind(bindings, body, env, scope = Scope.AGG)
```

Comment:
We need an `AggScanProxy`?

Reply:
not right now. The IRBuilder has a mishmash of stuff that was built because it was needed for the lowerer, it's definitely not complete.

The IRBuilder also only uses one type env, so I'm hesitant to expand scan/agg stuff there until we've taken a closer look.